spec/core/ics-002-client-semantics/README.md

```diff
@@ -607,6 +661,39 @@ function submitMisbehaviourToClient(
     abortTransactionUnless(clientState.checkForMisbehaviour(clientMessage))
     // update state based on misbehaviour
     clientState.UpdateStateOnMisbehaviour(misbehaviour)
+    // slash bond if any
+    bondAddress = address.Module(moduleName, subModuleName, id, "bonded_tokens_pool")
+    bank.iterateAccountBalances(bondAddress, (token: Coin) {
+        // slashed bond is paid out to the misbehaviour reporter
+        bank.sendCoinsFromModuleToAccount(bondAddress, misbehaviour.payee, token)
+    })
+    clear(bonded)
+}
+```
+
+#### Bonding
+
+A slashable bond may be placed on a client to economically guarantee correct behaviour and/or incentivize reporting of misbehaviour.
+
+```typescript
+function bondToken(
```

Comment:
We could do that 😄 ! Listing out some pros/cons off the top of my head.

Cons of bonding logic within light clients:
- lose standard RPC calling interface at the client level
- clients would need to expose a gRPC server with an arbitrary calling interface (can try to encourage a calling standard)
- no calling interface from `app` protocols (e.g. transfer and bond in single packet/tx)
  - maybe not a huge concern because they can always transfer first and then call RPC to bond in two separate tx
- reward distribution from utilization will be more complicated (not in the scope of this spec)
  - this issue should be more clearer when looking at client update incentivization

Pros of bonding logic within light clients:
- less complexity on 02-client side
- not every client requires a bond to be secure
- give light client writer more freedom

As a hybrid approach, slashing conditions can be pushed down into the client implementation but bond management could be handled within the client module?

Reply:
A different idea:
- Add `IBCClientMiddleware` for Client lifecycle methods
- Allow `IBCClientMiddleware` to register gRPC service/query routes